packages/contracts-bedrock/src/L2/CrossL2Inbox.sol

```diff
@@ -131,6 +122,23 @@ contract CrossL2Inbox is ICrossL2Inbox, ISemver, TransientReentrancyAware {
         emit ExecutingMessage(keccak256(_message), _id);
     }
 
+    /// @notice Validates a cross chain from on the destination chain. This function is useful
+    ///         for applications that understand the schema of the _message payload and want to
+    ///         process it in a custom way.
+    function validateMessage(Identifier calldata _id, bytes memory _message) external nonReentrant {
+        // Check the Identifier.
+        _checkIdentifier(_id);
+
+        emit ExecutingMessage(keccak256(_message), _id);
+    }
```

Comment:
```suggestion
    function validateMessage(Identifier calldata _id, bytes32 _msgHash) external nonReentrant {
        // Check the Identifier.
        _checkIdentifier(_id);

        emit ExecutingMessage(_msgHash, _id);
    }
```
since we don't really need the entire message for executing :)

Reply:
Ah good catch

Reply:
This is a very helpful optimization that could save lots of gas over the long run